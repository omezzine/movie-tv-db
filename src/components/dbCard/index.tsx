import constants from '../../constants';
import { Movie } from '../../models/movie';
import noImagePlaceHolder from '../../assets/No-Image-Placeholder.svg.png';
import './dbCard.scss';
import { DbType } from '../../models/db';
import { Tv } from '../../models/tv';

export interface OnCardClickEvent {
  id: number;
  type: DbType;
}
interface DbCardPropsI {
  type?: DbType;
  data: Movie & Tv;
  onClick: (obj: OnCardClickEvent) => void;
}

const DbCard: React.FC<DbCardPropsI> = ({ data, onClick, type }) => {
  const imagePath = data.backdrop_path
    ? `${constants.MOVIE_SERIE_DB.IMAGE_PATH}${data.backdrop_path}`
    : noImagePlaceHolder;
  return (
    <>
      <div className="card" onClick={(e) => onClick({ id: data.id, type: type || 'movie' })}>
        {<img className="card-img-top" src={imagePath} alt="Card image cap" />}
        <div className="card-body">
          <h5 className="card-title">{data.original_title || data.name}</h5>
          <p className="card-text text-truncate">{data.overview}</p>
        </div>
      </div>
    </>
  );
};

export default DbCard;
