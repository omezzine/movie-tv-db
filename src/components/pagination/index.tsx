

import { useState } from 'react';
import ReactPaginate from 'react-paginate';

export interface PaginationPropsI {
  offset?: number | string;
  pageCount: number;
  onChange: (newOffset: string) => void;
  currentPage?: string;
}
interface PageClickEvent {
    selected: number
}
const Pagination: React.FC<PaginationPropsI> = ({
  offset,
  pageCount,
  onChange,
  currentPage
}) => {
  const [, setItemOffset] = useState(offset);

  const handlePageClick = (event: PageClickEvent) => {
    onChange(String(event.selected + 1));
    setItemOffset(event.selected + 1);
  };

  return (
    <>
      <ReactPaginate
        nextLabel="next >"
        onPageChange={handlePageClick}
        forcePage={Number(currentPage || 1) - 1}
        pageRangeDisplayed={3}
        marginPagesDisplayed={2}
        pageCount={pageCount}
        previousLabel="< previous"
        pageClassName="page-item"
        pageLinkClassName="page-link"
        previousClassName="page-item"
        previousLinkClassName="page-link"
        nextClassName="page-item"
        nextLinkClassName="page-link"
        breakLabel="..."
        breakClassName="page-item"
        breakLinkClassName="page-link"
        containerClassName="pagination"
        activeClassName="active"
        renderOnZeroPageCount={undefined}
      />
    </>
  );
};
export default Pagination;