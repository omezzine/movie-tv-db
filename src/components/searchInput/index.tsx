import { QueryI } from '../../hooks/useSearchDb';
import { DbType } from '../../models/db';
import './style.scss';

interface SearchInputPropsI {
  placeHolder?: string;
  defaultValue: QueryI;
  onInputChange: (obj: QueryI) => void;
  onTypeChange: (obj: QueryI) => void;
}

const SearchInput: React.FC<SearchInputPropsI> = ({ placeHolder, onInputChange, onTypeChange, defaultValue }) => {
  return (
    <div className="search-input">
      <div className="input-group">
        <input
          defaultValue={defaultValue.query}
          onChange={(e) => onInputChange({ query: e.currentTarget.value })}
          type="text"
          className="form-control"
          placeholder={placeHolder}
          aria-label={placeHolder}
          aria-describedby="basic-addon2"
        />
        <div className="input-group-prepend">
          <select
            defaultValue={defaultValue.type || 'movie'}
            className="form-select"
            aria-label="Select Search Type"
            onChange={(e) => onTypeChange({ type: e.target.value as DbType, page: '1' })}
          >
            <option value="movie">Movie</option>
            <option value="tv">Serie</option>
          </select>
        </div>
      </div>
    </div>
  );
};

export default SearchInput;
