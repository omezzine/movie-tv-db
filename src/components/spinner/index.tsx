import  './spinner.scss';
export const Spinner = () => (
  <div role="alert" aria-busy="true" className='loader-container'>
    <div className='loader'></div>
  </div>
);