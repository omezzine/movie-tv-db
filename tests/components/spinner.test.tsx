import { render, screen } from '@testing-library/react';
import React from 'react';
import { Spinner } from '../../src/components/spinner';
import renderer from 'react-test-renderer';

test('should match snapshot spinner', async () => {
  const spinner = renderer.create(<Spinner />).toJSON();
  expect(spinner).toMatchSnapshot();
});

test('displays spinner', async () => {
  render(<Spinner />);
  await screen.findByRole('alert');
  expect(screen.getByRole('alert')).toBeDefined();
});
