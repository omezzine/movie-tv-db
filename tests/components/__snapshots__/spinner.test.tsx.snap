// Vitest Snapshot v1

exports[`should match snapshot spinner 1`] = `
<div
  aria-busy="true"
  className="loader-container"
  role="alert"
>
  <div
    className="loader"
  />
</div>
`;
